index.js

```diff
@@ -275,7 +263,22 @@ FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback) {
         );
       }
     }.bind(this));
-  }.bind(this))();
+  }.bind(this));
+
+  if (this._pendingWrites[path] === undefined) {
+    this._pendingWrites[path] = {
+      lastChange: now,
+      cancelWait: function() {
+        delete this._pendingWrites[path];
+        clearTimeout(timeoutHandler);
+        return callback();
```

Comment:
So that the pending [function](https://github.com/alexpusch/chokidar/blob/fix/waitWriteFinishTests/index.js#L156) will be invoked. I know it won't do anything since err and stat are null, but I still think we should not allow it to just hang there